### PR TITLE
cq.Shape and some free-func improvements

### PR DESCRIPTION
**Pull request by adam-urbanczyk**
- [x] `paramAt` for position [consistency]
- [x] `isSolid` fix [bugfix]
- [x] `solid` improvements [free-func]
- [x] `check` [free-func]
- [x] `innerShells` [consistency]
- [x] `outerShell` [consistency]
- [x] tests

### CODE REVIEW

**Comment by eguiraud-pf** [comment]
thank you @adam-urbanczyk , this would be enabling :)

**Comment by jons-pf** [comment]
@adam-urbanczyk Thanks for this PR!

File: cadquery/occ_impl/shapes.py [hunk 1]
```diff
@@ -2517,7 +2557,7 @@ def fillet(
     ) -> "Wire":
         """
         Apply 2D or 3D fillet to a wire
-
+        :param wire: The input wire to fillet. Currently only open wires are supported
```
Comment (lorenzncode) [inline cadquery/occ_impl/shapes.py:2560]: This docstring line was removed in #1630.   Probably got inadvertently added back.

**Review by lorenzncode — APPROVED** [review]
Looks good, no other issues found in limited testing.

File: cadquery/occ_impl/shapes.py (hunk 1 above)
Comment (adam-urbanczyk) [reply cadquery/occ_impl/shapes.py:2561]: Thanks, fixed!

**Review by jmwright — APPROVED** [review]
Looks good, thanks @adam-urbanczyk !

**Comment by adam-urbanczyk** [comment]
Alright, merging.

**Pull request merged**